neuralmonkey/decoders/multi_decoder.py

```diff
@@ -0,0 +1,134 @@
+import math
+import tensorflow as tf
+import numpy as np
+
+from neuralmonkey.vocabulary import START_TOKEN
+from neuralmonkey.decoding_function import attention_decoder
+from neuralmonkey.logging import log
+from neuralmonkey.dataset import Dataset
+
+
+class MultiDecoder(object):
+    """A class that manages parts of the computation graph that are
+    used for the decoding.
+    """
+
+    # pylint: disable=too-many-instance-attributes,too-many-locals
+    # Big decoder cannot be simpler. Not sure if refactoring
+    # it into smaller units would be helpful
+    # Some locals may be turned to attributes
+
+    def __init__(self, main_decoder, regularization_decoders, **kwargs):
+        """Create a new instance of the multi-decoder.
+
+        Arguments:
+            main_decoder: The decoder that corresponds to the output which
+                we want at runtime.
+
+            additional_decoders: A list of the decoders among which the multidecoder
+                will switch.
+
+        Keyword arguments:
+
+        """
+        self.main_decoder = main_decoder
+
+        self.regularization_decoders = regularization_decoders
+
+        self._training_decoders = [main_decoder] + regularization_decoders
+        self._decoder_costs = tf.concat(0, [tf.expand_dims(d.cost, 0)
+                                            for d in self._training_decoders])
+
+        self._scheduled_decoder = 0
+        self._input_selector = tf.placeholder(tf.float32,
```

Comment:
What is this for? (I mean, I can figure it out from the rest of the code, but it should be written here.)